source/commands/raffle/byrank.js

```diff
@@ -29,31 +30,34 @@ module.exports = new SubcommandWrapper("by-rank", "Select a user at or above a p
 						}))
 				)
 			],
-			flags: [MessageFlags.Ephemeral],
 			withResponse: true
-		}).then(response => response.resource.message.awaitMessageComponent({ time: 120000, componentType: ComponentType.StringSelect })).then(collectedInteraction => {
-			const rankIndex = Number(collectedInteraction.values[0]);
-			logicLayer.hunters.findHunterIdsAtOrAboveRank(interaction.guildId, rankIndex).then(qualifiedHunterIds => {
-				return interaction.guild.members.fetch({ user: qualifiedHunterIds });
-			}).then((unvalidatedMembers) => {
-				const eligibleMembers = unvalidatedMembers.filter(member => member.manageable);
-				if (eligibleMembers.size < 1) {
-					logicLayer.ranks.findAllRanks(interaction.guildId).then(ranks => {
-						const rank = ranks[rankIndex];
-						collectedInteraction.reply({ content: `There wouldn't be any eligible bounty hunters for this raffle (at or above the rank ${rank.roleId ? `<@&${rank.roleId}>` : `Rank ${rankIndex + 1}`}).`, flags: [MessageFlags.Ephemeral] });
-					});
-					return;
-				}
-				const winner = eligibleMembers.at(Math.floor(Math.random() * eligibleMembers.size));
-				collectedInteraction.reply(`The winner of this raffle is: ${winner}`);
-				logicLayer.companies.findCompanyByPK(interaction.guild.id).then(company => {
-					company.update("nextRaffleString", null);
-				});
-			})
+		}).then(message => message.awaitMessageComponent({ time: 120000, componentType: ComponentType.StringSelect })).then(async collectedInteraction => {
+			const varianceThreshold = Number(collectedInteraction.values[0]);
+			const reloadedRanks = await Promise.all(ranks.map(rank => rank.reload()));
+			const rankIndex = reloadedRanks.findIndex(rank => rank.varianceThreshold === varianceThreshold);
+			const rank = ranks[rankIndex];
```

Comment:
Since we're using the `rankIndex` from the `reloadedRanks`, shouldn't we be finding the rank by the reference in `reloadedRanks` instead? If the lists _are_ different there's a strong chance that the rank at that index is different in the new list.

Reply:
```suggestion
			const rank = reloadedRanks[rankIndex];
```
for easy fix application